komodo-relational/src/main/java/org/komodo/relational/teiid/CachedTeiid.java

```diff
@@ -204,6 +221,36 @@ public CachedTeiid resolve( final UnitOfWork transaction,
     Driver getDriver(final UnitOfWork transaction, String name) throws KException;
 
     /**
+     * Removes a Vdb from the cache, if present
+     * @param transaction
+     *        the transaction (cannot be <code>null</code> or have a state that is not {@link State#NOT_STARTED})
+     * @param name the Vdb name
+     * @throws KException
+     *         if an error occurs
+     */
+    void removeVdb(final UnitOfWork transaction, String name) throws KException;
+
+    /**
+     * Removes a dataSource from the cache, if present
+     * @param transaction
+     *        the transaction (cannot be <code>null</code> or have a state that is not {@link State#NOT_STARTED})
+     * @param name the dataSource name
+     * @throws KException
+     *         if an error occurs
+     */
+    void removeDataSource(final UnitOfWork transaction, String name) throws KException;
+
+    /**
+     * Removes a driver from the cache, if present
+     * @param transaction
+     *        the transaction (cannot be <code>null</code> or have a state that is not {@link State#NOT_STARTED})
+     * @param name the driver name
+     * @throws KException
+     *         if an error occurs
+     */
+    void removeDriver(final UnitOfWork transaction, String name) throws KException;
+
+    /**
      * @param uow
```

Comment:
In the strict sense of what a Cache should represent yes I agree with your comments.  But I want a way to force a Cache update without waiting for the designated refresh interval.  For example if a vdb is undeployed on the server, I know the Cache needs to remove that VDB.  Yes in a more generic way I could use a refreshVdbs method that would target only Vdbs.  But what if there are 500 VDBs deployed on the server?  Then undeployment of one server VDB would trigger refreshVDBs, reloading all 500 instead of the one affected.  This was a consideration.  I'm open to reworking this, but I want this scenario addressed.